### ISSUE TITLE
element.min.js

### ISSUE THREAD

**Issue by skurvish** [joomlahenk/fabrik#334]
I don't see this file in the original fabrik 3 codebase. Interestingly I do see it in my production system F3 but it lives in media/com_fabrik/js (not under dist). So I do not know how it comes to be under media if it does not exist in the repo.

@trob @joomlahenk  do you have any thoughts on this?

**Comment by trob** [comment]
It's not element-min.js but in media/fabrik/js all the minified files are in the subfolder /dist with their original name.
![grafik](https://user-images.githubusercontent.com/1045355/195972385-a98e89c7-0a4d-4569-a061-534260688538.png)
It's there in F3 GitHub and also here in F4 GitHub.

In plugins etc. it's still x.js and x-min.js files.

**Comment by joomlahenk** [comment]
It’s confusing, I agree.

Readable js files are in ...\media\com_fabrik\js and are called something.js

Minified js files are in in ...\media\com_fabrik\dist\js and are called something.js

 

So the minified element.js file is in ...\media\com_fabrik\dist\js

There is no element.min.js this files is called element.js

 

 

Van: skurvish ***@***.***> 
Verzonden: Saturday, 15 October 2022 03:12
Aan: joomlahenk/fabrik ***@***.***>
CC: joomlahenk ***@***.***>; Mention ***@***.***>
Onderwerp: [joomlahenk/fabrik] element.min.js (Issue #334)

 

I don't see this file in the original fabrik 3 codebase. Interestingly I do see it in my production system F3 but it lives in media/com_fabrik/js (not under dist). So I do not know how it comes to be under media if it does not exist in the repo.

@trob <https://github.com/trob>  @joomlahenk <https://github.com/joomlahenk>  do you have any thoughts on this?

**Comment by skurvish** [comment]
Ok, so here is what I have learned. 

The project build compress task (called using grunt js) scans recursively through all the plugins plus the administrator model and fields directories looking for all .js files, minifies them and stores them back into the same directory with the .min.js extension.

For files that live in media/com_fabrik/js it looks for all js in that directory specifically (i.e. non recursive), minifies them and stores them in media/com_fabrik/js/dist but without the min.js extension, just the js extension.

For files that live in media/com_fabrik/js/lib/datejs if does a recursive compress and store in place the minified files.

So, basically any js file that lives in media/com_fabrik/js root will be minified and stored in the dist subdirectory. 

When changing the js files in the above directory we will minify to dist without the min.js extension.

So in this case of element.js I will update that file in the dist directory and create a PR for it.

**Comment by skurvish** [comment]
@joomlahenk I see you updated the element.js in the dist folder. How did you minify the source js file?

**Comment by trob** [comment]
The minified files are named
xxx-min.js (not xxx.min.js)

**Comment by skurvish** [comment]
Sorry, that was simply a typo on my part.

**Comment by joomlahenk** [comment]
@skurvish To minify js, I use https://www.toptal.com/developers/javascript-minifier

**Comment by skurvish** [comment]
Best to use the grunt minifier when building packages as it inserts the release version in the minified file.

Can we close this?
